File: examples/test_mpi.c
#include <unistd.h>
#include <stdio.h>
#include <stdlib.h>
#include <mpi.h>
#include "include/Damaris.h"  

int client_code(int id) 
{
	float mydata[64][16][4];
	char* varname = "my group/my variable";
	char* event = "my script";

	int step;
	for(step = 0; step < 10; step++) {
	
	sleep(2);
	int i,j,k;
	for(i = 0; i < 64; i++) {
	for(j = 0; j < 16; j++) {
	for(k = 0; k <  4; k++) {
		mydata[i][j][k] = i*j*k;
	}
	}
	}

	DC_write(varname,step,mydata);
	DC_signal(event,step);

	}

	return 0;
}    

int main(int argc, char** argv) 
{
	MPI_Comm all = MPI_COMM_WORLD;
	MPI_Comm clients;
	int rank, size;

	if(argc != 2) {
		printf("Usage: mpirun -np 2 ./test_mpi_c <config.xml>\n");
		exit(0);
	}
	
	MPI_Init(&argc,&argv);

	if(DC_start_mpi_entity(argv[1],all)) {
		clients = DC_get_clients_communicator();
		MPI_Comm_rank(clients,&rank);
		MPI_Comm_size(clients,&size);
		client_code(rank);
		DC_kill_server();
		MPI_Barrier(clients);
		DC_finalize();
	}

	MPI_Finalize();
	return 0;
}

File: examples/test_mpi.f90
      program test
      include 'mpif.h'
      integer :: ierr
      integer :: comm
      integer :: is_client

      call MPI_INIT(ierr)
      comm = MPI_COMM_WORLD
      call df_start_mpi_entity("mpi.xml",comm,is_client)
      if(is_client.GT.0) then
         call client_code()
         call df_kill_server(ierr)
         call df_finalize(ierr)
      endif
      call MPI_FINALIZE(ierr)

      end program test

      subroutine client_code()
      integer :: ierr
      real, dimension(0:63,16:31,0:3) :: mydata
      integer :: i,j,k,step

      do step = 0, 10

      do i =  0, 63
      do j = 16, 31
      do k =  0,  3
          mydata(i,j,k) = i*j*k
      end do
      end do
      end do

      call df_write("my group/my variable",1,mydata,ierr)
      call df_signal("my event",1,ierr)
   
      end do

      end subroutine
